### PR TITLE
[13.0][IMP] product_assortment: Allow to use assortment based on partner domain

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT30584
ping @carlosdauden @joao-p-marques

### CODE REVIEW

File: product_assortment/models/ir_filters.py [hunk 1]
```diff
@@ -39,6 +43,25 @@ def _get_default_is_assortment(self):
     record_count = fields.Integer(compute="_compute_record_count")
 
     is_assortment = fields.Boolean(default=lambda x: x._get_default_is_assortment())
+    partner_domain = fields.Text(default="[]", required=True)
+    all_partner_ids = fields.Many2many(
```
Comment (joao-p-marques) [inline product_assortment/models/ir_filters.py:47]: Currently this is never seen by the user. Is it intended? Or could another button be added like this?
image

**Review by joao-p-marques — COMMENTED** [review]
Code and functional review. Just a small doubt, otherwise looks good :+1:

File: product_assortment/models/ir_filters.py (hunk 1 above)
Comment (sergio-teruel) [reply product_assortment/models/ir_filters.py:47]: Yes @joao-p-marques This is only a technical field to be use in other modules like as sale_order_product_assortment
Maybe we could add the smartbutton to open related partners

Comment (carlosdauden) [reply product_assortment/models/ir_filters.py:47]: The domain widget shows a button to access the records, so to make it homogeneous, perhaps the current smartbutton can be removed.

**Comment by pedrobaeza** [comment]
@joao-p-marques @carlosdauden please finish the review

**Review by carlosdauden — APPROVED** [review]

**Review by joao-p-marques — APPROVED** [review]

**Pull request merged**